MIGRATION.md

```diff
@@ -449,21 +461,128 @@ In 7.0, frameworks also specify the builder to be used. For example, The current
 - `@storybook/web-components-vite`
 - `@storybook/web-components-webpack5`
 
-We will be expanding this list over the course of the 7.0 development cycle. More info on the rationale here: [Frameworks RFC](https://www.notion.so/chromatic-ui/Frameworks-RFC-89f8aafe3f0941ceb4c24683859ed65c).
+You can find more info on the rationale here: [Frameworks RFC](https://chromatic-ui.notion.site/Frameworks-RFC-89f8aafe3f0941ceb4c24683859ed65c).
 
-#### frameworkOptions renamed
+**After picking your framework, you'll need to install it as a dev dependency.**
 
-In 7.0, the `main.js` fields `reactOptions` and `angularOptions` have been renamed. They are now options on the `framework` field:
+Because the new framework package will include the builder as well, you can remove any of the builder packages you were using before:
 
 ```js
-module.exports = {
+'@storybook/builder-webpack5',
+'@storybook/manager-webpack5',
+'@storybook/builder-webpack4',
+'@storybook/manager-webpack4',
+'@storybook/builder-vite',
+'storybook-builder-vite',
+```
+
+> Note:
+> if your project is still using Webpack 4, you'll have to upgrade to Webpack 5 as [Webpack 4 support was discontinued](#webpack4-support-discontinued)
+
+##### Framework field mandatory
+
+In 6.4 we introduced a new `main.js` field called [`framework`](#mainjs-framework-field). Starting in 7.0, the `main.js` file has to include a `framework` field and it should be of the package you picked in earlier steps.
+
+Here's an example, in case you picked `@storybook/react-vite`:
+
+```js
+// .storybook/main.js
+export default {
+  // ... your configuration
+  framework: {
+    name: '@storybook/react-vite',
+    options: {},
+  },
+};
+```
+
+##### frameworkOptions renamed
+
+In 7.0, the `main.js` fields `reactOptions` and `angularOptions` have been renamed. They are now options on the `framework` field.
+
+For React, what used to be:
+
+```js
+export default {
+  reactOptions: { fastRefresh: true },
+  framework: {
+    name: '@storybook/react-webpack5',
+    options: {},
+  },
+};
+```
+
+Becomes:
+
+```js
+export default {
   framework: {
     name: '@storybook/react-webpack5',
     options: { fastRefresh: true },
   },
 };
 ```
 
+For Angular, what used to be:
+
+```js
+export default {
+  angularOptions: { enableIvy: true },
+  framework: {
+    name: '@storybook/angular',
+    options: {},
+  },
+};
+```
+
+Becomes:
+
+```js
+export default {
+  framework: {
+    name: '@storybook/angular',
+    options: { enableIvy: true },
+  },
+};
+```
+
+##### builderOptions renamed
+
+In 7.0, the `main.js` fields `core.builder` are now removed, in favor of the new frameworks api. The builder is defined as part of the framework package you pick, e.g. `@storybook/vue3-vite`. If you had options for your builder, they are now options on the `framework.builder` field.
+
+What used to be:
+
+```js
+export default {
+  core: {
+    builder: {
+      name: 'webpack5',
```

Comment:
@yannbf so is `core.builder` completely ignored in SB7?

Reply:
No it isn't. SB reads `framework.options.builder` first, and if not there, it reads `core.builder.options`. You can see the logic [here](https://github.com/storybookjs/storybook/blob/feat/configDir-automigrations/code/lib/core-common/src/utils/get-builder-options.ts#L7)

I'd argue it's better to just tell people to move their builder stuff to framework, and have all users follow that pattern, with the only exception of a non-standard user config, which would use a third party builder or something that won't match the framework somehow. In that case though, the user probably would not follow this particular instruction as the first step is to pick a framework from the list of known frameworks we provide